ha/core/system_health/system_health.py

```diff
@@ -439,6 +439,9 @@ def process_event(self, healthevent: HealthEvent):
             if current_health:
                 current_health_dict = json.loads(current_health)
                 specific_info = current_health_dict["events"][0]["specific_info"]
+                # Get specific_info data from consul
+                if specific_info and specific_info.get('functional_type') and not healthevent.specific_info.get('functional_type'):
```

Comment:
What if this specific_info is blank? (HARE or CSM alert will not have anything in specific_info). But we need functional_type in specific_info compulsory because event_manager publish will be dependent on that.
In this case, it will not reach inside if.

Reply:
#L443
> specific_info.get('functional_type') 

add a default value None, so it should not break the code, if specific_info has no attribute 'functional_type'
 specific_info.get('functional_type', None)

Reply:
Python3 doesnot required to pass second argument for default value None. 
Python 3.6.8 (default, May 19 2021, 03:00:47)
[GCC 8.4.1 20200928 (Red Hat 8.4.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {1:2}
>>> a.get(5)
>>> g = a.get(5)
>>> print(g)
None
>>>

Reply:
Lets consider the situation. Hare subscribed to POD events for functional_type node and server.
Monitor sends event to system health and that event won't have functional_type. 
In this case, functional type should be determined. @gauri-bhosale

Reply:
Did you check "event without functional type" getting published?

Reply:
updated Test log with event without functional_type for disk mock_monitor please check @mariyappanp

Reply:
@Madhura-08  as mini provisioning already set functional_type so current_health capture specific_info functional_type we are just adding into health_event.specific_info that info, If specific_info is empty for other than node component(disk/cvg) specific_info will not added functional_type as we are setting functional_type for node component only. @ArchanaLimaye  please check if you have other opinion

Reply:
Ok..

Reply:
Gauri, can you please test Node event without giving functional_type and .add those testing details also?

Reply:
@gauri-bhosale 
In ha_setup, pod events are subscribed with functional_type ["server", "data"].
If functional_type is not present in pod event, then it won't be published. Component won't get event message.

Reply:
Got it. Thanks.